test/bats/kafka/kafka.bats

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bats
+
+# tags: kafka
+
+BATS_LIBS_PATH="${LIBS_PATH:-../libs}" # fallback to relative path if not set
+load "${BATS_LIBS_PATH}/bats-assert/load"
+load "${BATS_LIBS_PATH}/bats-support/load"
+load '../setup.bats'
+
+setup_file() {
+    export IONOS_TOKEN=$(ionosctl token generate)
+    mkdir -p /tmp/bats_test
+
+    run ionosctl datacenter create --location "de/fra" --name "cli-test-$(randStr 6)" -o json 2> /dev/null
+    assert_success
+
+    datacenter_id=$(find_or_create_resource \
+        "ionosctl datacenter list -M 1 -F location=de/fra,state=available -o json 2> /dev/null | jq -r '.items[] | .id'" \
+        "ionosctl datacenter create --name \"CLI-Test-$(randStr 8)\" --location de/fra -o json 2> /dev/null | jq -r '.id'")
+    [ -n "$datacenter_id" ] || fail "datacenter_id is empty"
+    assert_regex "$datacenter_id" "$uuid_v4_regex"
+    echo "$datacenter_id" > /tmp/bats_test/datacenter_id
+    retry_until "ionosctl datacenter get --datacenter-id ${datacenter_id} -o json 2> /dev/null | jq -r '.metadata.state'" \
+        "[[ \$output == \"AVAILABLE\" ]]" 10 60
+
+    lan_id=$(find_or_create_resource \
```

Comment:
This is more of a comment, and a personal anecdote, I found that using `find_or_create_resource` actually makes the test much more unpredictable, hard to debug, hard to read, hard to maintain etc.

You can obviously continue using it, but I have stopped using it for above reasons.

Reply:
Hmmm, I see, maybe we could look into removing it altogether from tests, in another PR of course. I will remove it from these tests for now.